x/stake/app_test.go

```diff
@@ -22,81 +20,89 @@ var (
-	mapp.Router().AddRoute("stake", NewHandler(keeper))
+	coinKeeper := bank.NewKeeper(mApp.AccountMapper)
+	keeper := NewKeeper(mApp.Cdc, keyStake, coinKeeper, mApp.RegisterCodespace(DefaultCodespace))
 
-	mapp.SetEndBlocker(getEndBlocker(keeper))
-	mapp.SetInitChainer(getInitChainer(mapp, keeper))
+	mApp.Router().AddRoute("stake", NewHandler(keeper))
+	mApp.SetEndBlocker(getEndBlocker(keeper))
+	mApp.SetInitChainer(getInitChainer(mApp, keeper))
 
-	require.NoError(t, mapp.CompleteSetup([]*sdk.KVStoreKey{keyStake}))
-	return mapp, keeper
+	require.NoError(t, mApp.CompleteSetup([]*sdk.KVStoreKey{keyStake}))
+	return mApp, keeper
 }
 
-// stake endblocker
+// getEndBlocker returns a stake endblocker.
 func getEndBlocker(keeper Keeper) sdk.EndBlocker {
 	return func(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
 		validatorUpdates := EndBlocker(ctx, keeper)
+
 		return abci.ResponseEndBlock{
 			ValidatorUpdates: validatorUpdates,
 		}
 	}
 }
 
-// overwrite the mock init chainer
+// getInitChainer initializes the chainer of the mock app and sets the genesis
+// state. It returns an empty ResponseInitChain.
 func getInitChainer(mapp *mock.App, keeper Keeper) sdk.InitChainer {
 	return func(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
 		mapp.InitChainer(ctx, req)
+
 		stakeGenesis := DefaultGenesisState()
 		stakeGenesis.Pool.LooseTokens = 100000
+
 		InitGenesis(ctx, keeper, stakeGenesis)
 
 		return abci.ResponseInitChain{}
 	}
 }
 
-//__________________________________________________________________________________________
-
-func checkValidator(t *testing.T, mapp *mock.App, keeper Keeper,
-	addr sdk.Address, expFound bool) Validator {
-
+func checkValidator(
+	t *testing.T, mapp *mock.App, keeper Keeper,
+	addr sdk.Address, expFound bool,
+) Validator {
 	ctxCheck := mapp.BaseApp.NewContext(true, abci.Header{})
 	validator, found := keeper.GetValidator(ctxCheck, addr1)
+
 	require.Equal(t, expFound, found)
 	return validator
 }
 
-func checkDelegation(t *testing.T, mapp *mock.App, keeper Keeper, delegatorAddr,
-	validatorAddr sdk.Address, expFound bool, expShares sdk.Rat) {
-
+func checkDelegation(
+	t *testing.T, mapp *mock.App, keeper Keeper, delegatorAddr,
+	validatorAddr sdk.Address, expFound bool, expShares sdk.Rat,
+) {
 	ctxCheck := mapp.BaseApp.NewContext(true, abci.Header{})
 	delegation, found := keeper.GetDelegation(ctxCheck, delegatorAddr, validatorAddr)
 	if expFound {
 		require.True(t, found)
-		assert.True(sdk.RatEq(t, expShares, delegation.Shares))
+		require.True(sdk.RatEq(t, expShares, delegation.Shares))
+
 		return
 	}
+
 	require.False(t, found)
 }
 
 func TestStakeMsgs(t *testing.T) {
-	mapp, keeper := getMockApp(t)
+	mApp, keeper := getMockApp(t)
 
-	genCoin := sdk.Coin{"steak", sdk.NewInt(42)}
-	bondCoin := sdk.Coin{"steak", sdk.NewInt(10)}
+	genCoin := sdk.Coin{Denom: "steak", Amount: sdk.NewInt(42)}
```

Comment:
We've actually been tending towards removing the named fields, especially when theres many structs being created in a row like a test table. 

That said, would prob be best, at least for these, to use the sdk.NewCoin constructor

Reply:
Mhmmm, is this for tests only? If not, I think it'd be better to use a constructor as you mentioned. Otherwise, there is no clear indication of what these parameters mean (i.e. `f := Foo{1,1,3,true,4}`.

Reply:
Yes, primarily for tests where we're initializing many test cases.

In code, ideally we use the field names when creating struct literals.

In bot cases, better to use a constructor.

Reply:
Sounds good 👍